oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/UserInfoRequestSchema.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.core;
+
+import java.io.Serializable;
+
+import org.springframework.security.core.SpringSecurityCoreVersion;
+import org.springframework.util.Assert;
+
+/**
+ * This section defines three methods of sending bearer access tokens in resource requests to resource servers.
+ *  Clients MUST NOT use more than one method to transmit the token in each request.
+ *
+ * <p>
+ * The OAuth 2.0 Authorization Framework defines three standard request schema types:
+ * Authorization Request Header Field, Form-Encoded Body Parameter and URI Query Parameter.
+ * It also provides an extensibility mechanism for defining additional request schema types.
+ *
+ * Because of the security weaknesses associated with the URI method,
+ * including the high likelihood that the URL containing the access token will be logged,
+ * it SHOULD NOT be used unless it is impossible to transport the access token in the
+ * "Authorization" request header field or the HTTP request entity-body.
+ * Resource servers MAY support this method.
+ *
+ * @author MyeongHyeon Lee
+ * @since 5.1
+ * @see <a target="_blank" href="https://tools.ietf.org/html/rfc6750#section-2">Section 2 Authenticated Requests</a>
+ */
+public final class UserInfoRequestSchema implements Serializable {
+	private static final long serialVersionUID = SpringSecurityCoreVersion.SERIAL_VERSION_UID;
+	public static final UserInfoRequestSchema HEADER = new UserInfoRequestSchema("header");
+	public static final UserInfoRequestSchema FORM = new UserInfoRequestSchema("form");
```

Comment:
Can you also add `QUERY` but don't implement/use it in the `OAuth2UserService`. This will come in handy in Resource Server feature.